modules/dists/BlockDist.chpl

```diff
@@ -1724,7 +1724,7 @@ proc BlockArr.doiScan(op, dom) where (rank == 1) &&
 
   // The result of this scan, which will be Block-distributed as well
   type resType = op.generate().type;
-  var res: [dom] resType;
+  var res: [dom] resType = noinit;
```

Comment:
@mppf we chatted about this before break, but that was long enough ago that I have forgotten -- I think this works correctly for POD types, but is this wrong for non-POD? I think we discussed something about manually calling the array creation routine `initElts=false` or something, but I cannot remember more than that.

Reply:
Right, you can call `dom.buildArray(initElts=false)` to allow non-POD but you have to move initialize the elements if they are records.  Here I think that happens in chpl__postScan.

Reply:
If I wanted to only optimize for POD types (which i imagine will be the type for almost all scans) does what I have work? Or do I need to do something like:


```chpl
var res = dom.buildArray(resType, initElts=!isPOD(resType));
...
if !isPOD(resType) then res.dsiElementInitializationComplete();
```

Reply:
You'll need something like that, yes. Otherwise you will get compilation errors on the array `noinit` used with a non-POD type.

Reply:
Changed to that. Think this is ready for a review.